permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java

```diff
@@ -306,25 +311,25 @@ void requestPermissions(
     @PermissionConstants.PermissionStatus
     private int determinePermissionStatus(
             @PermissionConstants.PermissionGroup int permission,
-            Context context) {
+            Activity activity) {
```

Comment:
This change might result into issue because the `activity` might be `null` in certain conditions (e.g. when the app is running in the background). 

I noticed this method uses the activity a few times without making sure it is not `null` which will result into runtime exceptions. I think we should provide an alternative solution, maybe using casting where we validate if the supplied `context` is of type `Activity` and if so do more advanced status validation, if not stick to the current implementation.